### PR TITLE
Docs: Update release notes and documentation for 1.6 release

### PR DESCRIPTION
**Pull request by shefty**
Signed-off-by: Sean Hefty <sean.hefty@intel.com>

### CODE REVIEW

**Comment by shefty** [comment]
This commit pulls in updates to the readme file from the v1.5.x branch.  I added placeholders for provider updates.

@hppritcha @jswaro @j-xiong @a-ilango @gladkovdmitry17 @aingerson -- Please either send me updates to include, or open a PR with additional updates.  I will attempt to start a list based on parsing git log messages as well.

**Comment by hppritcha** [comment]
obviously ignore the LANL CI checker.  having problems with the single remaining Cray instance with all systems at NERSC are down.

**Comment by shefty** [comment]
Details for psm, psm2, sockets, and verbs have been updated.

**Comment by jsquyres** [comment]
@shefty I added a commit with a usNIC update for NEWS.md.  Feel free to squash into the previous NEWS commit.

**Comment by shefty** [comment]
thanks

**Comment by shefty** [comment]
Squashed Jeff's patch into mine.  Added a new commit that updates the man pages on handling manual progress.  @j-xiong @epaulson10

File: man/fi_endpoint.3.md [hunk 1]
```diff
@@ -187,7 +187,12 @@ bound to one or more fabric resources.  An endpoint that will generate
 asynchronous completions, either through data transfer operations or
 communication establishment events, must be bound to the appropriate
 completion queues or event queues, respectively, before being enabled.
-Unconnected endpoints must be bound to an address vector.
+Additionally, endpoints that use manual progress must be associated
+with relevant completion queues or event queues in order to drive
+progress.  For endpoints that are only used as the target of RMA or
+atomic operations, this means binding the endpoint to a completion
+queue associated with receive processing.  Unconnected endpoints must
```
Comment (j-xiong) [inline man/fi_endpoint.3.md]: Binding a counter should also work for the purpose of making manual progress.

Comment (shefty) [reply man/fi_endpoint.3.md]: I thought about this.  We don't allow an endpoint to be bound to a counter without also being bound to a completion queue on the initiator side.  I'm not sure we want to allow that case at the target side.  The current text for binding to a counter says, "An endpoint may also be bound to a fabric counter", though that was a recent clarification.

**Comment by shefty** [comment]
@a-ilango - I added commit with rxm release note details.  Please review.

**Pull request merged**